### ISSUE TITLE
Error with pihole image "s6-rc-compile: fatal: unable to read"

### ISSUE THREAD

**Issue by dannykorpan** [jacklul/pihole-updatelists#120]
Hi,

my pihole container won't start after pulling and starting the latest image version from jacklul/pihole:latest "41aa699aab8346e7a9a32574d9ec168fc54490cc" with error
`s6-rc-compile: fatal: unable to read /etc/s6-overlay/s6-rc.d/_gravityonboot/type: No such file or directory`
Also deleting the volumes to flush configs and recreating the container outputs the same error.

Using a Raspberry Pi 4 with DietPi 8.10.2 and latest docker version.

Any hints?

Kind Regards,
Danny

**Comment by jacklul** [comment]
Something must have changed in the image and it broke the integration, I will look it into it.

In the meantime you can either pull official Pi-hole image or one of the previous tags https://hub.docker.com/r/jacklul/pihole/tags which shouldn't have the issue.

**Comment by dannykorpan** [comment]
Hi,

there are no more docker logs. How do I get the logs?

Kind Regards,
Danny

**Comment by jacklul** [comment]
I believe you can start the container attached to console window and then just copy the output from console directly.

**Comment by seanob86** [comment]
I got this error and there was only 1 line output after issuing this cmd ```
docker logs -f pihole
```

`s6-rc-compile: fatal: unable to read /etc/s6-overlay/s6-rc.d/_gravityonboot/type: No such file or directory `

**Comment by jacklul** [comment]
I found the issue already by viewing changes in the Pi-hole image, a fix will come later today.

**Comment by jacklul** [comment]
**latest** and **2022.11** images should now be working

**Comment by dannykorpan** [comment]
I can confirm it's working again.

```
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service cron: starting
s6-rc: info: service cron successfully started
s6-rc: info: service _uid-gid-changer: starting
s6-rc: info: service _uid-gid-changer successfully started
s6-rc: info: service _startup: starting
  [i] Starting docker specific checks & setup for docker pihole/pihole
  [i] Setting capabilities on pihole-FTL where possible
  [i] Applying the following caps to pihole-FTL:
        * CAP_CHOWN
        * CAP_NET_BIND_SERVICE
        * CAP_NET_RAW
  [i] Ensuring basic configuration by re-running select functions from basic-install.sh
  [i] Installing configs from /etc/.pihole...
  [i] Existing dnsmasq.conf found... it is not a Pi-hole file, leaving alone!
  [i] Installing /etc/dnsmasq.d/01-pihole.conf...
  [✓] Installed /etc/dnsmasq.d/01-pihole.conf
  [i] Installing /etc/.pihole/advanced/06-rfc6761.conf...
  [✓] Installed /etc/dnsmasq.d/06-rfc6761.conf
  [i] Installing latest logrotate script...
  [✓] Installing latest logrotate script
  [i] Creating empty /etc/pihole/setupVars.conf file.
  [i] Assigning password defined by Environment Variable
  [✓] New password set
  [i] Added ENV to php:
			"TZ" => "Europe/Berlin",
			"PIHOLE_DOCKER_TAG" => "",
			"PHP_ERROR_LOG" => "/var/log/lighttpd/error-pihole.log",
			"CORS_HOSTS" => "",
			"VIRTUAL_HOST" => "0.0.0.0",
  [i] Using IPv4 and IPv6
  [i] setup_blocklists now setting default blocklists up: 
  [i] TIP: Use a docker volume for /etc/pihole/adlists.list if you want to customize for first boot
  [i] Blocklists (/etc/pihole/adlists.list) now set to:
https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts
  [i] Setting DNS servers based on PIHOLE_DNS_ variable
Resolved unbound from PIHOLE_DNS_ as: 172.24.0.2
  [i] Applying pihole-FTL.conf setting LOCAL_IPV4=0.0.0.0
  [i] FTL binding to default interface: eth0
  [i] Enabling Query Logging
  [i] Testing lighttpd config: Syntax OK
  [i] All config checks passed, cleared for startup ...
  [i] Docker start setup complete
  [i] pihole-FTL (no-daemon) will be started as pihole
s6-rc: info: service _startup successfully started
s6-rc: info: service pihole-FTL: starting
s6-rc: info: service pihole-FTL successfully started
s6-rc: info: service _updatelistsonboot: starting
s6-rc: info: service lighttpd: starting
s6-rc: info: service _updatelistsonboot successfully started
s6-rc: info: service lighttpd successfully started
s6-rc: info: service _postFTL: starting
'/etc/pihole-updatelists.conf' -> '/etc/pihole-updatelists/pihole-updatelists.conf'
s6-rc: info: service _postFTL successfully started
  Checking if custom gravity.db is set in /etc/pihole/pihole-FTL.conf
ownership of '/etc/pihole-updatelists/pihole-updatelists.conf' retained as root:root
s6-rc: info: service legacy-services: starting
mode of '/etc/pihole-updatelists/pihole-updatelists.conf' retained as 0644 (rw-r--r--)
Lists update skipped - gravity database not found
s6-rc: info: service legacy-services successfully started
  [i] Creating new gravity database
  [i] Migrating content of /etc/pihole/adlists.list into new database
  [i] Neutrino emissions detected...

  [✓] Pulling blocklist source list into range
  [i] Preparing new gravity database...
  [✓] Preparing new gravity database
  [i] Using libz compression
  [i] Target: https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts
  [i] Status: Pending...
  [✓] Status: Retrieval successful
  [i] Analyzed 158531 domains
  [i] Creating new gravity databases...
  [✓] Creating new gravity databases
  [i] Storing downloaded domains in new gravity database...
  [✓] Storing downloaded domains in new gravity database
  [i] Building tree...
  [✓] Building tree
  [i] Swapping databases...
  [✓] Swapping databases
  [✓] The old database remains available.
  [i] Number of gravity domains: 158531 (158531 unique domains)
  [i] Number of exact blacklisted domains: 0
  [i] Number of regex blacklist filters: 0
  [i] Number of exact whitelisted domains: 0
  [i] Number of regex whitelist filters: 0
  [i] Cleaning up stray matter...
  [✓] Cleaning up stray matter
  [✓] FTL is listening on port 53
     [✓] UDP (IPv4)
     [✓] TCP (IPv4)
     [✓] UDP (IPv6)
     [✓] TCP (IPv6)
  [i] Pi-hole blocking will be enabled
  [i] Enabling blocking

  [✓] Pi-hole Enabled
  Pi-hole version is v5.14.1 (Latest: v5.14.1)
  AdminLTE version is v5.17 (Latest: v5.17)
  FTL version is v5.19.1 (Latest: v5.19.1)
  Container tag is: 2022.11
```
